--- origsrc/ExtUtils-F77-1.16/F77.pm	2007-04-01 23:40:23.000000000 -0500
+++ src/ExtUtils-F77-1.16/F77.pm	2009-10-25 03:36:21.714105800 -0500
@@ -198,22 +198,6 @@ $F77config{Generic}{G77}{Cflags} = '-O';
 $F77config{Generic}{DEFAULT} = 'G77';
 $F77config{Generic}{F2c}     = $F77config{Generic}{G77};
 
-### cygwin ###
-#"-lg2c -lm";
-# needed this on my cygwin system to get things working properly
-sub getcyglink {
-   return join ' ', map {my $lp = `g77 -print-file-name=lib$_.a`;
-			$lp =~ s|/[^/]+$||;
-			 $lp =~ s|L([a-z,A-Z]):|L//$1|g;
-			 "-L$lp -l$_"} qw/g2c m/;
-}
-
-$F77config{Cygwin}{G77}{Trail_} = 1;
-$F77config{Cygwin}{G77}{Compiler} = 'g77';
-$F77config{Cygwin}{G77}{Cflags} = '-O';
-$F77config{Cygwin}{G77}{Link}	= \&getcyglink;
-$F77config{Cygwin}{DEFAULT}	= 'G77';
-
 ### Linux ###
 
 $F77config{Linux}{G77}     = $F77config{Generic}{G77};
@@ -256,6 +240,13 @@ if (ucfirst($Config{'osname'}) eq "Linux
   # code for ifort, g95 etc........
 }
 
+### Cygwin ###
+
+$F77config{Cygwin}{GFortran} = $F77config{Linux}{GFortran};
+$F77config{Cygwin}{G77}      = $F77config{Generic}{G77};
+$F77config{Cygwin}{F2c}      = $F77config{Generic}{G77};
+$F77config{Cygwin}{DEFAULT}  = 'GFortran';
+
 ### DEC OSF/1 ###
 
 $F77config{Dec_osf}{F77}{Link}   = "-L/usr/lib -lUfor -lfor -lFutil -lm -lots -lc";
